### PR TITLE
URL Inference

### PR DESCRIPTION
**Pull request by jeff-hernandez**
This will infer the release tag to determine which repository to use. 

### Output Sample

```
HEAD is now at ee905ab bump version
WARNING: The directory '/github/home/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.

==================================================
Release v0.0.0.dev1 was published on GitHub
==================================================

warning: check: missing required meta-data: url

WARNING: The directory '/github/home/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: The directory '/github/home/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.

----------  Uploading to Test PyPI  --------------------

Uploading distributions to https://test.pypi.org/legacy/
Uploading featuretools_tsfresh_primitives-0.0.0.dev1-py3-none-any.whl

  0%|          | 0.00/53.1k [00:00<?, ?B/s]
100%|██████████| 53.1k/53.1k [00:01<00:00, 33.5kB/s]
Uploading featuretools_tsfresh_primitives-0.0.0.dev1.tar.gz

  0%|          | 0.00/22.5k [00:00<?, ?B/s]
100%|██████████| 22.5k/22.5k [00:01<00:00, 16.1kB/s]

View at:
https://test.pypi.org/project/featuretools-tsfresh-primitives/0.0.0.dev1/
 ```

### CODE REVIEW

**Review by rwedge — COMMENTED** [review]
For the version parsing we could try using the [packaging library](https://pypi.org/project/packaging/)

It has a [version parser](https://packaging.pypa.io/en/latest/version/#) that has a `is_dev_release` flag we could use

**Comment by rwedge** [comment]
We are pretty careful with official releases about the tag matching the package version but with dev versions people might not be as careful.  It could be good to add a check where, after we checkout the tagged version of the repo, we run `python setup.py -V` to get the version in the code and compare it to the tag

File: upload.sh [hunk 1]
```diff
@@ -1,44 +1,102 @@
...
+if isinstance(version, Version):
+    if version.is_devrelease:
+        print('Test PyPI')
 
-upload_to_pypi () {
-    # Checkout specified tag
-    git checkout tags/$tag
+    if not version.is_devrelease \
+        and not version.is_postrelease \
+        and not version.is_prerelease:
```
Comment (jeff-hernandez) [inline upload.sh]: Do we want to release post or pre releases to PyPI?

Comment (rwedge) [reply upload.sh]: I think it's more likely that we want to do a pre or post release to the real PyPI servers than a dev release, so we could reserve dev releases as the type that is automatically uploaded to the test server and the other are uploaded to the real server.

Comment (jeff-hernandez) [reply upload.sh]: Okay, updated script.

File: upload.sh [hunk 2]
```diff
@@ -1,44 +1,101 @@
...
+pip install --quiet packaging==19.2
+
+match=$(python -c "
+from packaging.version import parse
+
+match = parse('$tag') == parse('$version')
+print(match)
+")
+
+if [ $match = "False" ]; then
```
Comment (rwedge) [inline upload.sh]: The logic here is currently "do not release if $match evaluates to False" but I think better logic would be "do  not release if $match is not True"

That way if something goes wrong and match is neither "True" or "False" we stop the action

Edit: or add a third case if $match is neither true nor false

Comment (jeff-hernandez) [reply upload.sh]: Sounds good. I updated the logic to exit if match is not true.

File: upload.sh [hunk 3]
```diff
@@ -1,44 +1,101 @@
...
+# Get release tag from environment and checkout branch.
 tag=$(basename $GITHUB_REF)
+git checkout tags/$tag
 
-# Get action that triggered release event
+# Get package version.
+version=$(python setup.py --version)
+
+# Check if release tag matches the package version.
+pip install --quiet packaging>=17.0
```
Comment (rwedge) [inline upload.sh]: `packaging>=17.0` needs to be enclosed in quotes so the `>` is not interpreted as a command to redirect the output

Comment (jeff-hernandez) [reply upload.sh]: I see. Updated to enclose in quotes.

**Review by rwedge — APPROVED** [review]
Looks good!

**Pull request merged**